repositories/src/main/scala/cromwell/pipeline/datastorage/dto/FileParameter.scala

```diff
@@ -0,0 +1,43 @@
+package cromwell.pipeline.datastorage.dto
+
+import org.mongodb.scala.Document
+import play.api.libs.json._
+
+final case class FileParameter(name: String, typedValue: TypedValue)
+
+object FileParameter {
+  import OptionFormat._
+
+  implicit object InputNodeFormat extends Format[FileParameter] {
+    override def writes(o: FileParameter): JsValue = {
+      val (valType, value) = TypedValue.toPair(o.typedValue)
+      JsObject(
+        Seq(
+          "name" -> JsString(o.name),
+          "type" -> JsString(valType),
+          "value" -> (value match {
```

Comment:
Are you sure you need `null` in case of absent value?
Even if you do, your `OptionFormat` object already implements this logic, you should do it it manually here.

Reply:
`OptionFormat` is used there to use `(json \ "value").as[Option[String]]`, don't know how to implement logic of `OptionFormat.writes`